fargate/log-puller.yml

```diff
@@ -205,6 +217,64 @@ Resources:
       ListenerArn:
         Fn::ImportValue:
           !Join [':', [!Ref 'StackName', 'PublicListener443']]
-      Priority: !Ref 'Priority'
+      Priority: !Ref 'PriorityForPath'
+
+  LoadBalancerRule80Host:
+    Type: AWS::ElasticLoadBalancingV2::ListenerRule
+    Properties:
+      Actions:
+        - TargetGroupArn: !Ref 'TargetGroup'
+          Type: 'forward'
+      Conditions:
+        - Field: host-header
+          Values:
+            - Fn::Join:
+                - ''
+                - - !Ref Subdomain
+                  - '.'
+                  - Fn::ImportValue: !Join [':', [!Ref 'StackName', 'HostedZoneName']]
+      ListenerArn:
+        Fn::ImportValue:
+          !Join [':', [!Ref 'StackName', 'PublicListener80']]
+      Priority: !Ref 'PriorityForHostname'
+  LoadBalancerRule443Host:
+    Type: AWS::ElasticLoadBalancingV2::ListenerRule
+    Properties:
+      Actions:
+        - TargetGroupArn: !Ref 'TargetGroup'
+          Type: 'forward'
+      Conditions:
+        - Field: host-header
+          Values:
+            - Fn::Join:
+                - ''
+                - - !Ref Subdomain
+                  - '.'
+                  - Fn::ImportValue: !Join [':', [!Ref 'StackName', 'HostedZoneName']]
+      ListenerArn:
+        Fn::ImportValue:
+          !Join [':', [!Ref 'StackName', 'PublicListener443']]
+      Priority: !Ref 'PriorityForHostname'
+
+  DNSRecord:
+    Type: AWS::Route53::RecordSet
```

Comment:
One thing I've learned to be useful is to use weighting on the record sets in Route53. Like this: https://github.com/concord-consortium/cloud-formation/blob/master/lara-ecs.yml#L169-L170

Without this if you try to refactor the template and put the record set in a sub template, the template will fail to deploy because the old record set will conflict with the new one.  But if you use weighting which requires the Id field. Then you can have overlapping record sets and CloudFront doesn't care. It also makes it possible to convert a manually created record set into one managed by cloudformation without downtime.

Reply:
Thanks - that will save me a few hairs on my head :)